Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -1214,12 +1214,15 @@ namespace AZ
     //=========================================================================
     void ComponentApplication::CreateStaticModules(AZStd::vector<AZ::Module*>& outModules)
     {
+        // Add this to the front of the list. In monolithic builds, dynamic modules become static modules,
+        // so we want this to load before the dynamic modules load regardless of the build type.
+        outModules.insert(outModules.begin(), aznew AzCoreModule());
```

Comment:
In addition to this change I recommend changing the static module initialization code in `ComponentApplication::LoadStaticModules`, to perform initialization of ALL modules in particular phase before moving on to next phase.

```c++
    void ComponentApplication::LoadStaticModules()
    {
        // Load static modules
        auto LoadStaticModules = [this](ModuleManagerRequests* moduleManager)
        {
            auto PopulateStaticModules = [this](AZStd::vector<AZ::Module*>& outModules) { this->CreateStaticModules(outModules); };

            // 1. Load - the first call is to ensure all dynamic modules are loaded
            // 2. CreateClass - the second call is to create specific AZ::Module class instances for each dynamic module after its loaded
            // 3. RegisterComponentDescriptors - the third call is to perform a horizontal register step for each module component descriptors after
            //    module has been loaded and created
            for (ModuleInitializationSteps lastStepToPerform : { ModuleInitializationSteps::Load,
                ModuleInitializationSteps::CreateClass, ModuleInitializationSteps::RegisterComponentDescriptors })
            {
                moduleManager->LoadStaticModules(PopulateStaticModules, lastStepToPerform);
            };
        };

        ModuleManagerRequestBus::Broadcast(LoadStaticModules);
    }

```